service/storage_proxy.cc

```diff
@@ -1727,7 +1729,7 @@ class abstract_write_response_handler : public seastar::enable_shared_from_this<
     }
     db::view::update_backlog max_backlog() {
         return std::ranges::fold_left(
-                get_targets() | std::views::transform([this] (locator::host_id ep) {
+                std::views::join(std::array{_targets, _successful_targets}) | std::views::transform([this] (locator::host_id ep) {
```

Comment:
Is it important to calculate the backlog after requests completes? Can we pre-calculate it when we create write response handler and use the value to calculate delay on completion?

Reply:
sounds good to me
I changed it to pre-calculate the backlog

Reply:
We could have marginally more accurate delays if we were calculating the backlog only after getting the responses (which carry the up-to-date backlogs from corresponding nodes). But if we were only using backlogs from the targets which didn't respond yet, it shouldn't make a big difference if we get the backlogs a bit earlier, and the addition of extra targets which we didn't take into account before should make up for that